Geometry/TrackerGeometryBuilder/src/TrackerGeomBuilderFromGeometricDet.cc

```diff
@@ -101,26 +103,46 @@ TrackerGeometry* TrackerGeomBuilderFromGeometricDet::build(const GeometricDet* g
   // now building the Pixel-like subdetectors
   for (unsigned int i = 0; i < 6; ++i) {
     if (gdsubdetmap[i] == GeometricDet::PixelBarrel)
-      buildPixel(
-          dets[i], tracker, GeomDetEnumerators::SubDetector::PixelBarrel, false, BIG_PIX_PER_ROC_X, BIG_PIX_PER_ROC_Y);
+      buildPixel(dets[i],
+                 ptitpx,
+                 tracker,
+                 GeomDetEnumerators::SubDetector::PixelBarrel,
+                 false,
+                 BIG_PIX_PER_ROC_X,
+                 BIG_PIX_PER_ROC_Y);
```

Comment:
just for me to understand. Once you put the big pixel info in the `PTrackerAdditionalParametersPerDet`, what are these parameters `BIG_PIX_PER_ROC_X` and `BIG_PIX_PER_ROC_Y` going to be ?

Reply:
It depends on the module type:
- 3D sensors: BIG_PIX_PER_ROC_X = 0;  BIG_PIX_PER_ROC_Y = 0
- planar sensors 1x2:  BIG_PIX_PER_ROC_X = 0;  BIG_PIX_PER_ROC_Y = 1 
- planar sensors 2x2:  BIG_PIX_PER_ROC_X = 2;  BIG_PIX_PER_ROC_Y = 1

Reply:
I am not sure to follow. Does this imply duplication of information?
Currently:

https://github.com/cms-sw/cmssw/blob/a5b91a1d0848665cc6f5d3ddb11be28272302e1c/Geometry/TrackerGeometryBuilder/src/TrackerGeomBuilderFromGeometricDet.cc#L53-L54

Reply:
I agree that we need to avoid duplication, limiting the usage of boolean like "upgradeGeometry".

About duplication, I think that this is already the case for what concerns the number of rows and cols per ROC, as they are also defined in trackerParameters.xml (NB: with outdated phase-1 values)

https://github.com/cms-sw/cmssw/blob/a5b91a1d0848665cc6f5d3ddb11be28272302e1c/Geometry/TrackerCommonData/data/PhaseII/trackerParameters.xml#L4-L6

but AFAIK these values are never used in the phase-2 code.

Reply:
what I meant is that in the piece of code above `BIG_PIX_PER_ROC_X` and `BIG_PIX_PER_ROC_Y` are coming from `PTrackerParameters` which is itself an EventSetup product. In the new scheme of things will these numbers be stored in `PTrackerAdditionalParametersPerDet` instead? What about the values for non phase-2 geometries?

Reply:
> The idea is to store the information in PTrackerAdditionalParametersPerDet, but the details of the implementation, e.g. storing two numbers or instead two booleans with the actual mapping occurring in the class based on the number of ROC in X and in Y, is still under development.

to be honest, I would be reluctant to merge this revert without a clear plan on the implementation details for supplying the variable (per module) number of big pixel per ROC in the phase-2 geometries, at the risk of large duplication of information. I think this PR should be resubmitted once a more clear path is presented.

Reply:
I understand your point. My idea was to factorize the changes to make easier the development of the code for the BigPixel part.

If the changes related to the usage of  PTrackerAdditionalParametersPerDet are fine anyway (including the suggestions from @missirol below), it is ok to postpone the merge once also the other part of the code will be ready.